### PR TITLE
singlehtml: deprecate the 'fix_refuris' helper function

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Deprecation

### Purpose
- Deprecates a `singlehtml` builder helper method that I believe should be redundant since #11970 and #12551 are in place, but that might, perhaps, be re-used by other functionality (extensions / codebases).

### Detail
- Deprecates the `sphinx.builders.singlehtml.SingleFileHTMLBuilder.fix_refuris` method, and adds a warning noting that it's planned for removal in Sphinx 9 (the next major version).

### Relates
- Follow-up to #11970 and #12551.

Edit: update description: this PR removes calls to the method, so don't imply that this is deprecation of already-unused code.

### CODE REVIEW

**Review by AA-Turner — CHANGES_REQUESTED** [review]
Remember to add new deprecations to `doc/extdev/deprecated.rst`.

A

**Comment by jayaddison** [comment]
Although I'm fairly keen to merge this, and think it should be safe -- maybe I'll take one more self-review pass (and/or try some builds of external `singlehtml` projects) on it next week before re-requesting approval/merge.

File: CHANGES.rst [hunk 1]
```diff
@@ -11,6 +11,9 @@ Incompatible changes
 
 Deprecated
 ----------
+* #13037: Deprecate the ``SingleHTMLBuilder.fix_refuris`` method, with removal
+  planned for version 9.0 of Sphinx.
+  Patch by James Addison.
```
Comment (AA-Turner) [inline CHANGES.rst]:
```suggestion

* #13037: Deprecate the ``SingleHTMLBuilder.fix_refuris`` method.
  Patch by James Addison.
```

File: doc/extdev/deprecated.rst [hunk 2]
```diff
@@ -22,6 +22,11 @@ The following is a list of deprecated interfaces.
      - Removed
      - Alternatives
 
+   * - ``sphinx.builders.singlehtml.SingleFileHTMLBuilder.fix_refuris``
+     - 8.2
+     - 9.0
```
Comment (AA-Turner) [inline doc/extdev/deprecated.rst]:
```suggestion
     - 10.0
```

File: sphinx/builders/singlehtml.py [hunk 3]
```diff
@@ -2,11 +2,13 @@
 
 from __future__ import annotations
 
+import warnings
 from typing import TYPE_CHECKING, Any
 
 from docutils import nodes
 
 from sphinx.builders.html import StandaloneHTMLBuilder
+from sphinx.deprecation import RemovedInNextVersionWarning
```
Comment (AA-Turner) [inline sphinx/builders/singlehtml.py]:
```suggestion
from sphinx.deprecation import RemovedInSphinx10Warning
```

File: sphinx/builders/singlehtml.py [hunk 4]
```diff
@@ -52,6 +54,14 @@ def get_relative_uri(self, from_: str, to: str, typ: str | None = None) -> str:
         return self.get_target_uri(to, typ)
 
     def fix_refuris(self, tree: Node) -> None:
+        deprecation_msg = (
+            "The 'SingleFileHTMLBuilder.fix_refuris' method is no longer used "
+            'within the builder and is planned for removal in Sphinx 9. '
+            'Please report malformed URIs generated by the Sphinx singlehtml '
+            'builder as bugreports.'
+        )
+        warnings.warn(deprecation_msg, RemovedInNextVersionWarning, stacklevel=2)
```
Comment (AA-Turner) [inline sphinx/builders/singlehtml.py]:
```suggestion
        deprecation_msg = (
            "The 'SingleFileHTMLBuilder.fix_refuris' method is no longer used "
            'within the builder and is planned for removal in Sphinx 10. '
            'Please report malformed URIs generated by the Sphinx singlehtml '
            'builder as bugreports.'
        )
        warnings.warn(deprecation_msg, RemovedInSphinx10Warning, stacklevel=2)
```

**Review by AA-Turner — COMMENTED** [review]
Our deprecation policy calls for two major releases. Any reason to expidite this one?

A

**Comment by jayaddison** [comment]
> Our deprecation policy calls for two major releases. Any reason to expidite this one?

Nope, this can wait until Sphinx 10 - oddly I think I configured that originally; I can't remember why I reset that to Sphinx 9.

**Review by AA-Turner — APPROVED** [review]

**Pull request merged**